### PR TITLE
refactor: Convert server.rs to use async await

### PR DESCRIPTION
**Pull request by drahnr**
mozilla#787

### CODE REVIEW

**Review by TriplEight — APPROVED** [review]
Also can be merged as it was tested in https://github.com/mozilla/sccache/pull/787

File: src/server.rs [hunk 1]
```diff
@@ -587,14 +588,13 @@ impl<C: CommandCreatorSync> SccacheServer<C> {
...
-            .block_on(Timeout::new(Compat::new(wait), Duration::new(30, 0)))
-            .map_err(|e| {
-                if e.is_inner() {
-                    e.into_inner().unwrap()
-                } else {
-                    io::Error::new(io::ErrorKind::Other, e)
-                }
-            })?;
+            .block_on_std(async {
+                time::timeout(Duration::new(30, 0), wait)
```
Comment (gww-parity) [inline src/server.rs:592]: does it indicate waiting 30 seconds?

If yes please update info log at line 579

File: src/server.rs [hunk 2]
```diff
@@ -544,35 +548,32 @@ impl<C: CommandCreatorSync> SccacheServer<C> {
...
+            Ok(())
+        }));
 
-        let server = future::select_all(futures);
-        runtime.block_on(server).map_err(|p| p.0)?;
+        let server = future::select_all(futures).map(|t| t.0);
+        runtime.block_on_std(server)?;
 
         info!(
             "moving into the shutdown phase now, waiting at most 10 seconds \
```
Comment (gww-parity) [inline src/server.rs:579]: herer to update to 30 seconds , following code review comment below

**Review by gww-parity — CHANGES_REQUESTED** [review]

File: src/server.rs (hunk 2 above)
Comment (drahnr) [inline src/server.rs:579]:
```suggestion
            "moving into the shutdown phase now, waiting at most 30 seconds \
```

Comment (gww-parity) [inline src/server.rs:579]: 30 seconds (according to line 592)

**Review by gww-parity — APPROVED** [review]

**Review by gww-parity — APPROVED** [review]

**Pull request merged**